src/main/java/nl/tudelft/cse1110/andy/execution/process/ExternalProcess.java

```diff
@@ -0,0 +1,79 @@
+package nl.tudelft.cse1110.andy.execution.process;
+
+import java.io.IOException;
+import java.util.Scanner;
+
+public class ExternalProcess extends Thread {
```

Comment:
It's usually more common to implement `Runnable` rather than extending `Thread`, but I guess the final result is the same.

Reply:
I moved the run method to an inner class since it's only used for reading output. The ExternalProcess class no longer has to extend `Thread` (or implement `Runnable`). The inner class implements `Runnable`.